### PR TITLE
add cancel pipeline run selection to pipeline runs list

### PR DESCRIPTION
**Pull request by steveodonovan**
# Changes
https://github.com/tektoncd/dashboard/issues/53

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/components/CancelButton/CancelButton.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  handleShow = () => {
+    this.setState({ showDialog: true });
+  };
+
+  render() {
+    const closeIcon = <StopFilled16 />;
+    const { showDialog } = this.state;
+    const { type, name } = this.props;
+    return (
+      <>
```
Comment (AlanGreene) [inline src/components/CancelButton/CancelButton.js]: Carbon provides a ModalWrapper component which wraps a button, modal, and associated logic
http://react.carbondesignsystem.com/?path=/story/modalwrapper--transactional-passive-modal

File: package.json [hunk 2]
```diff
@@ -53,6 +53,8 @@
     "babel-loader": "^8.0.5",
     "clean-webpack-plugin": "^2.0.1",
     "css-loader": "^2.1.1",
+    "enzyme": "^3.9.0",
```
Comment (AlanGreene) [inline package.json]: Do we really need enzyme? Was there a particular issue that couldn't be resolved with react-testing-library alone?

File: src/components/CancelButton/CancelButton.js (hunk 1 above)
Comment (AlanGreene) [reply src/components/CancelButton/CancelButton.js]: ignore this, ModalWrapper doesn't work well with icon triggers

**Comment by steveodonovan** [comment]
/hold cancel

@AlanGreene Decided to include taskrun cancel in the list view in the same PR

File: src/containers/PipelineRuns/PipelineRuns.js [hunk 3]
```diff
@@ -224,6 +227,20 @@ export /* istanbul ignore next */ class PipelineRuns extends Component {
                       : ''}
                   </StructuredListCell>
                   <StructuredListCell>{lastTransitionTime}</StructuredListCell>
+                  <StructuredListCell>
+                    {isRunning(reason, status) && (
+                      <CancelButton
+                        type="Pipeline Run"
```
Comment (AlanGreene) [inline src/containers/PipelineRuns/PipelineRuns.js]: should probably be 'PipelineRun' and 'TaskRun' (no space) for consistency with other places in the UI and to match the CRD kinds.

File: src/containers/TaskRunList/TaskRunList.js [hunk 4]
```diff
@@ -84,13 +86,14 @@ export /* istanbul ignore next */ class TaskRunList extends Component {
...
             <StructuredListCell head>Last Transition Time</StructuredListCell>
+            <StructuredListCell head />
           </StructuredListRow>
         </StructuredListHead>
         <StructuredListBody>
           {!taskRuns.length && (
             <StructuredListRow>
               <StructuredListCell>
-                <span>No pipeline runs</span>
+                <span>No task runs</span>
```
Comment (AlanGreene) [inline src/containers/TaskRunList/TaskRunList.js]: 👍

File: src/components/CancelButton/CancelButton.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+.cancel-button {
+  background: none;
+  color: inherit;
+  border: none;
+  padding: 0;
+  font: inherit;
+  cursor: pointer;
+  outline: inherit;
+  fill: #ad1625;
```
Comment (AlanGreene) [inline src/components/CancelButton/CancelButton.scss]: is there a colour token that's a match or good alternative?

File: src/components/CancelButton/CancelButton.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+          type="button"
+          className="cancel-button"
+          onClick={this.handleShow}
+        >
+          {closeIcon}
+        </button>
+        <Modal
+          open={showDialog}
+          modalHeading={`Stop ${type}`}
+          primaryButtonText="Yes"
```
Comment (AlanGreene) [inline src/components/CancelButton/CancelButton.js]: After looking at this UI in action I have a request:

- let's mirror the heading in the primary button (e.g. 'Stop PipelineRun')
- and use 'Cancel' for the secondary button

This gives us consistency with other modals in the UI, and makes it clearer* what clicking each button does.

\* also need to move the title from the icon to the button (to get the tooltip to display consistently), and change the text to 'Stop {type}'

File: src/components/CancelButton/CancelButton.scss (hunk 5 above)
Comment (steveodonovan) [reply src/components/CancelButton/CancelButton.scss]: Went for the same as is used by the failed icon

File: src/components/CancelButton/CancelButton.stories.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import { text } from '@storybook/addon-knobs';
+import StoryRouter from 'storybook-react-router';
+
+import CancelButton from './CancelButton';
+
+storiesOf('CancelButton', module)
+  .addDecorator(StoryRouter())
+  .add('default', () => (
+    <CancelButton
+      type={text('Run Type', 'Pipeline Run')}
```
Comment (AlanGreene) [inline src/components/CancelButton/CancelButton.stories.js]: PipelineRun

File: src/components/CancelButton/CancelButton.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  };
+
+  handleShow = () => {
+    this.setState({ showDialog: true });
+  };
+
+  render() {
+    const { showDialog } = this.state;
+    const { type, name } = this.props;
+    const closeIcon = <StopFilled16 title={`Stop ${type}`} />;
```
Comment (AlanGreene) [inline src/components/CancelButton/CancelButton.js]: left over from https://github.com/tektoncd/dashboard/pull/301#discussion_r300071749

move the `title` attribute from the icon to the button so it's displayed consistently (very difficult to trigger it on the icon at the moment)

**Pull request merged**